### ISSUE TITLE
How to handle out of picture keypoints with YOLO pose

### ISSUE THREAD

**Issue by ArnauCampanera** [ultralytics/ultralytics#19506]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hi!

I am trying to use YOLO pose estimation to track bear poses in camera trap images. After a very tedious annotation process, I found myself getting the "out of boundaries" error, as in many pictures you can't see the whole animal. Because of that, some keypoints were left out of the picture, and getting [1.XXX] or negative coordinates.

From here, what solutions do I have? I read somewhere that yolo uses all the keypoints, no matter if they are not visible, and visibility is just used for loss function or something like that? 

If I normalize every out of the picture keypoint and they end up in random places of the image, what impact will this have in the model if they are marked non-visible? Will the model still learn something (wrong) from them?

Is there a way to ignore non-visible keypoints?

Thank you very much.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @ArnauCampanera, thank you for your question and for using Ultralytics 🚀! It sounds like you're working on an exciting pose estimation project with YOLO! 

We recommend consulting our [Docs](https://docs.ultralytics.com/) for detailed information on YOLO pose estimation, where you might find helpful guidance for handling keypoints and visibility. Specifically, the [model training tips](https://docs.ultralytics.com/guides/model-training-tips/) section might provide insights into managing challenging datasets.

If this is a ❓ Question, as it seems, please provide as much supporting information as possible (e.g., an example of your dataset annotations, a description of your training setup, or any relevant logs). This will help the team better understand the scenario and provide more specific guidance.

If you believe this could be related to a 🐛 Bug, we encourage you to share a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) illustrating your issue. Providing this will help us investigate and potentially resolve any unexpected behavior more effectively.

### Key Recommendations:
- Ensure your `ultralytics` package is up-to-date by upgrading in your Python environment:
```bash
pip install -U ultralytics
```
- Double-check your environment dependencies ([Python>=3.8](https://www.python.org/), [PyTorch>=1.8](https://pytorch.org/get-started/locally/)) as mismatches here could impact performance.

### Community & Support:
You're invited to join the broader Ultralytics community for real-time discussion and support:
- 🗨️ [Discord](https://discord.com/invite/ultralytics) for live chat
- 🧵 [Discourse](https://community.ultralytics.com/) for detailed discussions
- 📋 [Subreddit](https://reddit.com/r/Ultralytics) for sharing insights with other users

### Environment Options:
While investigating or testing solutions, you can use our verified environments with all pre-installed dependencies:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Cloud platforms**, see the guides for [Google Cloud](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/) or [AWS](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/).
- **Docker Images** are available for streamlined deployment on local or cloud servers: <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

### Ongoing Support:
This is an automated response to help set you on the right path 😊. An Ultralytics engineer will review your issue and provide further assistance as soon as possible. 

Let us know if you encounter any further challenges or have additional details to share in the meantime!

**Comment by 666git-sys** [comment]
请问如何用yolo11关键点检测模型预测后的人体照片通过程序来判断照片中的每个人体是正面还是侧面呢，并在程序中输出每张照片里面每个对象的判断结果

**Comment by glenn-jocher** [comment]
@666git-sys for front/side orientation detection using YOLO11 pose estimation, analyze keypoint visibility and spatial relationships from the pose results. Specifically:

1. Use the `Keypoints.conf` attribute to check visibility scores (visible when >0.5) 
2. Calculate shoulder/hip keypoint angles using `Keypoints.xy` coordinates
3. Implement logic: Front-facing typically shows symmetrical shoulder/hip visibility and alignment, while side views exhibit unilateral occlusion and depth-axis displacement

Example code snippet for orientation analysis:
```python
results = model.predict(source)
for r in results:
    kpts = r.keypoints.xyn.cpu().numpy()  # normalized keypoints
    vis = r.keypoints.conf.cpu().numpy()  # visibility scores
    # Add orientation logic using keypoint positions/visibility
```

For keypoint index references, see the Pose Estimation Datasets documentation.
